Automattic/jetpack
# Sync: improve importer detection and sync start action

Take two on the work started in #11523. See D25761-code.

#### Changes proposed in this Pull Request:

* Adds a utility function (`get_calling_importer_class`) which infers the import class responsible for calling the import action.
* Reworks the logic for triggering the sync action so that it is the same for `import_start`, `import_done`, and `import_end`.

#### Testing instructions:





* Go to `/wp-admin/admin.php?import=wordpress`
* Complete an import. If the import completes successfully, two actions should be synced
  * `jetpack_sync_import_start`
  * `jetpack_sync_import_end`
* Both events should have two parameters (`wordpress`, `WordPress`)
* Other types of imports that trigger the `import_done` or `import_end` actions should also sync `jetpack_sync_import_done`. Examples: (Blogger, LiveJournal, RSS). Each should have two parameters, and as long as the importer is in `known_importers`, those parameters should reflect those values.  These other importers, however, don't currently trigger `import_start`, so `jetpack_sync_import_start` won't be triggered.

#### Proposed changelog entry for your changes:


No changelog entry needed
